configuration.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/charmbracelet/bubbles/key"
+)
+
+type LlamaConfiguration struct {
+	Bindings []BindingConfiguration
+}
+
+type BindingConfiguration struct {
+	Action   string
+	Keys     []string
+	Disabled bool
+	Help     key.Help
+}
+
+var (
+	keyMap = map[string]*key.Binding{
+		"keyForceQuit": &keyForceQuit,
+		"keyQuit":      &keyQuit,
+		"keyOpen":      &keyOpen,
+		"keyBack":      &keyBack,
+		"keyUp":        &keyUp,
+		"keyDown":      &keyDown,
+		"keyLeft":      &keyLeft,
+		"keyRight":     &keyRight,
+		"keyTop":       &keyTop,
+		"keyBottom":    &keyBottom,
+		"keyLeftmost":  &keyLeftmost,
+		"keyRightmost": &keyRightmost,
+		"keyVimUp":     &keyVimUp,
+		"keyVimDown":   &keyVimDown,
+		"keyVimLeft":   &keyVimLeft,
+		"keyVimRight":  &keyVimRight,
+		"keyVimTop":    &keyVimTop,
+		"keyVimBottom": &keyVimBottom,
+		"keySearch":    &keySearch,
+		"keyPreview":   &keyPreview,
+		"keyDelete":    &keyDelete,
+		"keyUndo":      &keyUndo,
+	}
+)
+
+func processConfig() {
+	file, err := os.Open(getConfigPath())
+	if err != nil {
+		return // Don't load configuration
+	}
+	defer file.Close()
+
+	configBytes, err := io.ReadAll(file)
+	if err != nil {
+		panic(err)
+	}
+
+	var llamaConfig LlamaConfiguration
+	err = json.Unmarshal(configBytes, &llamaConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, bindingConfig := range llamaConfig.Bindings {
+		if binding, exists := keyMap[bindingConfig.Action]; exists {
+			binding.SetKeys(bindingConfig.Keys...)
+			binding.SetHelp(bindingConfig.Help.Key, bindingConfig.Help.Desc)
+			binding.SetEnabled(!bindingConfig.Disabled)
+		}
+	}
+}
+
+func getConfigPath() string {
+	// Try to resolve path from environment variable
+	value, ok := os.LookupEnv("LLAMA_CONFIG")
```

Comment:
There's already a lookup function in `main.go`, try to reuse it

Reply:
Ah good catch, thank you!

I'm taking a look at the `main.go lookup` function now, but it looks like it was constructed to accommodate a specific use case around the dual `LLAMA_EDITOR` and `EDITOR` environment variables. I'd like to refactor this a bit because for most use cases involving environment variables we'll only be trying to resolve one, and it feels clunky to pass single-element arrays to the function.

Something like this is what I have in mind:
```go
func lookup(name string, defaultValue string) string {
    value, ok := os.LookupEnv(name)
    if ok && value != "" {
        return value
    }
    return defaultValue
}

func lookupFirst(names []string, defaultValue string) string {
    for _, name := range names {
        value := lookupt(name, "")
        if value != "" {
            return value
        }
    }
    return defaultValue
}
```

Reply:
Looks good to me!